controllers/front/paymentreturn.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+* Team Paylike
+*
+*  @author    Team Paylike
+*  @copyright Team Paylike
+*  @license   MIT license: https://opensource.org/licenses/MIT
+*/
+
+class PaylikePaymentReturnModuleFrontController extends ModuleFrontController
+{
+	public function __construct()
+	{
+		parent::__construct();
+		$this->context = Context::getContext();
+	}
+
+	public function init()
+	{
+		parent::init();
+		$cart = $this->context->cart;
+		$paylike = new Paylike();
+		$total = $cart->getOrderTotal(true, Cart::BOTH);
+		$customer = new Customer((int)$cart->id_customer);
+		$currency = new Currency((int)$cart->id_currency);
+		$paylikeapi = new PaylikeAPI(Configuration::get('PAYLIKE_SECRET_KEY'));
+
+		if (Configuration::get('PAYLIKE_CHECKOUT_MODE') == 'delayed')
+		{
+			$capture = $paylikeapi->transactions->fetch(Tools::getValue('transactionid'),
+			[
+			'currency' => $currency->iso_code,
+			'amount' => Tools::ps_round($total, 2) * 100,
+			]);
```

Comment:
Sending it where? Looks like you're fetching the transaction.

I think what you meant to do was to check the authenticity of the transaction. To do that you should fetch the transaction, check the currency, amount and a unique order identifier with what's on record.